docs/schema-generator/execution/exceptions.md

```diff
@@ -3,21 +3,30 @@ id: exceptions
 title: Exceptions and Partial Data
 ---
 
+## Returning GraphQL Errors
+
 Exceptions thrown during execution of a query will result in a GraphQLError that is added to a list of errors of the result. See
 [graphql-java documentation](https://www.graphql-java.com/documentation/v14/execution/) for more details on how to customize your exception handling.
 
-### Partial Data
 
-GraphQL allows you to return both data and errors in a single response. Depending on the criticality of the encountered error you may want to return
-partial data together with the corresponding errors. In Kotlin, functions return only a single value, which means that in order to return both data
+```kotlin
+fun getRandomNumberOrError(): Int {
+    val num = Random().nextInt(100)
+    return if (num <= 50) num else throw Exception("number is greater than 50")
+}
+```
+
+## Returning Data and Errors
+
+GraphQL allows you to return both data and errors in a single response, as long as the data returned still matches the schema. Depending on the criticality of the encountered error you may want to return
+null for some fields with the corresponding errors. In Kotlin, functions return only a single value, which means that in order to return both data
```

Comment:
while previous sentence specifies that result has to match the schema i think this sentence might be somewhat confusing (i.e. nulls are only allowed for nullable fields). Maybe we should clarify that it should be some `default data or null for optional fields` or something like that